src/pytest_plugins/consume/simulators/simulator_logic/test_via_sync.py

```diff
@@ -0,0 +1,540 @@
+                if sync_peer_count > 0 or client_peer_count > 0:
+                    logger.debug(
+                        f"Peers connected: client_under_test={client_peer_count}, "
+                        f"sync_client={sync_peer_count}"
+                    )
+            except Exception as e:
+                logger.debug(f"Could not check peer count: {e}")
+
+        except Exception as e:
+            logger.warning(f"Failed to trigger sync with newPayload/forkchoice update: {e}")
+    else:
+        logger.warning(
+            f"Could not find payload for block {last_valid_block_hash} to send to sync client"
+        )
+
+    # Wait for synchronization with continuous forkchoice updates
+    with timing_data.time("Wait for synchronization"):
+        # Get the target block number for logging
+        target_block = eth_rpc.get_block_by_hash(last_valid_block_hash)
+        target_block_number = int(target_block["number"], 16) if target_block else "unknown"
+        logger.info(
+            f"Waiting for sync client to reach block #{target_block_number} "
+            f"(hash: {last_valid_block_hash})"
+        )
+
+        # Start monitoring sync progress
+        sync_start_time = time.time()
+        last_sync_block_number = 0
+        sync_check_interval = 1.0
+        last_check_time = time.time()
+        last_forkchoice_time = time.time()
+        forkchoice_interval = 2.0  # Send forkchoice updates every 2 seconds
+
+        while time.time() - sync_start_time < 10:  # 10 second timeout
+            # Send periodic forkchoice updates to keep sync alive
+            if time.time() - last_forkchoice_time >= forkchoice_interval:
+                try:
+                    # Send forkchoice update to sync client to trigger/maintain sync
+                    assert sync_engine_rpc is not None, "sync_engine_rpc is required"
+                    sync_fc_response = sync_engine_rpc.forkchoice_updated(
+                        forkchoice_state=ForkchoiceState(
+                            head_block_hash=last_valid_block_hash,
+                            safe_block_hash=last_valid_block_hash,
+                            finalized_block_hash=fixture.genesis.block_hash,
+                        ),
+                        payload_attributes=None,
+                        version=fixture.sync_payload.forkchoice_updated_version
+                        if fixture.sync_payload
+                        else fixture.payloads[-1].forkchoice_updated_version,
+                    )
+                    status = sync_fc_response.payload_status.status
+                    logger.debug(f"Periodic forkchoice update status: {status}")
+                    last_forkchoice_time = time.time()
+                except Exception as fc_err:
+                    logger.debug(f"Periodic forkchoice update failed: {fc_err}")
+
+            # Check sync progress periodically
+            if time.time() - last_check_time >= sync_check_interval:
```

Comment:
This second block feels to me that should be outside of the loop once we got "VALID" from the forkchoice updated (no need to check multiple times if we are still getting "SYNCING", once we get "VALID" it's only a one-time check I think).